src/Renderer/ThreeExtended/Feature2Mesh.js

```diff
@@ -57,263 +57,326 @@ function fillColorArray(colors, offset, length, r, g, b) {
+
+        coordinatesToVertices(vert, extrude, vertices, 3 * vert.length);
+        fillColorArray(colors, vert.length, color[1].r * 255, color[1].g * 255, color[1].b * 255, vert.length);
     }
-    geometry.addAttribute('position', new THREE.BufferAttribute(vertices, 3));
-    geometry.addAttribute('color', new THREE.BufferAttribute(colors, 3, true));
-    return new THREE.Points(geometry);
+
+    const geom = new THREE.BufferGeometry();
+    geom.addAttribute('position', new THREE.BufferAttribute(vertices, 3));
+    geom.addAttribute('color', new THREE.BufferAttribute(colors, 3, true));
+    return geom;
 }
 
-function coordinateToLines(coordinates, properties, options) {
-    const indices = [];
-    const vertices = new Float32Array(3 * coordinates.coordinates.length);
-    const colors = new Uint8Array(3 * coordinates.coordinates.length);
-    const geometry = new THREE.BufferGeometry();
-    let offset = 0;
-
-    /* eslint-disable-next-line */
-    for (const id in coordinates.featureVertices) {
-        const { contour, property } = extractFeature(coordinates, properties, id);
-        // get altitude from properties
-        const altitude = getAltitude(options, property, contour);
-        coordinatesToVertices(contour, altitude, vertices, offset * 3);
-
-        // set indices
-        const line = coordinates.featureVertices[id];
-        // TODO optimize indices lines
-        // is the same array each time
-        for (let i = line.offset; i < line.offset + line.count - 1; ++i) {
-            indices.push(i);
-            indices.push(i + 1);
-        }
 
-        // assign color to each point
-        const color = getColor(options, property);
-        fillColorArray(colors, offset, contour.length, color.r * 255, color.g * 255, color.b * 255);
+function geometryToPoint(geometry, properties, options, multiGeomAttributes) {
+    const vertices = multiGeomAttributes ? multiGeomAttributes.vertices : geometry.vertices;
 
-        // increment offset
-        offset += contour.length;
-    }
+    // get altitude / color from properties
+    const altitude = getAltitude(options, properties, vertices);
+    const color = getColor(options, properties);
+
+    const geom = prepareBufferGeometry(
+        vertices,
+        color,
+        altitude);
 
-    geometry.addAttribute('position', new THREE.BufferAttribute(vertices, 3));
-    geometry.addAttribute('color', new THREE.BufferAttribute(colors, 3, true));
-    geometry.setIndex(new THREE.BufferAttribute(new Uint32Array(indices), 1));
-    return new THREE.LineSegments(geometry);
+    return new THREE.Points(geom);
 }
 
-function coordinateToPolygon(coordinates, properties, options) {
+function geometryToLine(geometry, properties, options, multiGeomAttributes) {
+    const vertices = multiGeomAttributes ? multiGeomAttributes.vertices : geometry.vertices;
+
+    // get altitude / color from properties
+    const altitude = getAltitude(options, properties, vertices);
```

Comment:
In `globe_wfs_extruded`, it seems that `altitude` resolves as an array, making everything a `NaN` afterwards.

Reply:
My bad, I was working with a previous version. All good for this !